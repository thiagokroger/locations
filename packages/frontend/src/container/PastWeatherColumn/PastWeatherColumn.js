import React from 'react'
import PropTypes from 'prop-types'
import { WeatherCard } from '../../components'

const PastWeatherColumn = ({ pastSearch, removeSearchValue }) => {
  return (
    <div className="column is-one-quarter">
      <span className="column-title">Last searched places</span>
      {
        pastSearch ? pastSearch.map((value, i) =>
          <WeatherCard
            key={i}
            clime={value.weatherDescriptions}
            observationTime={value.utc}
            temperature={value.temperature}
            city={value.cityName}
            removeSearchValue={removeSearchValue}
          />) : null
      }
    </div>
  )
}

PastWeatherColumn.propTypes = {
  pastSearch: PropTypes.array,
  removeSearchValue: PropTypes.func.isRequired
}

PastWeatherColumn.defaultProps = {
  pastSearch: []
}

export default PastWeatherColumn
