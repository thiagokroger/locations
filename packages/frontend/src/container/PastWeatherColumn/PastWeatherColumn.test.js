import React from 'react'
import { shallow } from 'enzyme'
import PastWeatherColumn from './PastWeatherColumn'

describe('Containers - Past Weather Column without not required props', () => {
  it('should render without crashing', () => {
    const component = shallow(
      <PastWeatherColumn removeSearchValue={jest.fn()} />
    )
    expect(component).toMatchSnapshot()
  })
})

describe('Containers - Past Weather Column with all props', () => {
  it('should render without crashing', () => {
    const component = shallow(
      <PastWeatherColumn removeSearchValue={jest.fn()} pastSearch={['Search Value test']} />
    )
    expect(component).toMatchSnapshot()
  })
})
