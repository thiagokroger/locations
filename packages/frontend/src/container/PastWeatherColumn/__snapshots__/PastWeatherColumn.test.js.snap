// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers - Past Weather Column with all props should render without crashing 1`] = `
<div
  className="column is-one-quarter"
>
  <span
    className="column-title"
  >
    Last searched places
  </span>
  <WeatherCard
    city=""
    clime={Array []}
    key="0"
    observationTime=""
    removeSearchValue={[MockFunction]}
    temperature={0}
  />
</div>
`;

exports[`Containers - Past Weather Column without not required props should render without crashing 1`] = `
<div
  className="column is-one-quarter"
>
  <span
    className="column-title"
  >
    Last searched places
  </span>
</div>
`;
